### DISCUSSION TITLE
Coverage report

### DISCUSSION THREAD

**Discussion by rubenochiavone** [solidusio/solidus#4553] [New Features or Ideas]
Solidus has an extensive test spec, but there is no report about code coverage.

Currently it is possible to know each sub-project code coverage report by doing

```bash
$ COVERAGE=true ./bin/build
```

However, this information in divided between `rspec`s executions and doesn't report overall coverage.

I'd suggest

- Add code coverage report to whole project (not only for each sub-project)
- Add code coverage badge to `README.md`
- Add code coverage report for each PR

WDYT?

PS.: I did some work in advance but I'd rather discuss it before opening a PR - see https://github.com/rubenochiavone/solidus/tree/coverage

**Comment by kennyadsl** [comment]
@rubenochiavone I'm all for this. I think it's a good idea. I'm not sure if we really need to create another CircleCi job for that. Can't we use an existing one? Coverage should be the same right?

**Comment by jacobherrington** [comment]
I'd love to see a PR that does this as well. Go for it!

**Comment by rubenochiavone** [comment]
Thanks for the feedback.

Before opening the PR let me ask you, is using [codecov.io](https://codecov.io) ok? If so, could you create a `CODECOV_TOKEN` to be used in the PR?

> I'm not sure if we really need to create another CircleCi job for that. Can't we use an existing one? Coverage should be the same right?

Definitively. We can improve on that during the PR
